sky/clouds/cudo.py

```diff
@@ -0,0 +1,314 @@
+
+    @classmethod
+    def regions_with_offering(cls, instance_type,
+                              accelerators: Optional[Dict[str, int]],
+                              use_spot: bool, region: Optional[str],
+                              zone: Optional[str]) -> List[clouds.Region]:
+        assert zone is None, 'Cudo does not support zones.'
+        del accelerators, zone  # unused
+        if use_spot:
+            return []
+
+        regions = service_catalog.get_region_zones_for_instance_type(
+            instance_type, use_spot, 'cudo')
+
+        if region is not None:
+            regions = [r for r in regions if r.name == region]
+        return regions
+
+    @classmethod
+    def get_vcpus_mem_from_instance_type(
+            cls,
+            instance_type: str,
+    ) -> Tuple[Optional[float], Optional[float]]:
+
+        return service_catalog.get_vcpus_mem_from_instance_type(instance_type,
+                                                                clouds='cudo')
+
+    @classmethod
+    def zones_provision_loop(
+            cls,
+            *,
+            region: str,
+            num_nodes: int,
+            instance_type: Optional[str] = None,
+            accelerators: Optional[Dict[str, int]] = None,
+            use_spot: bool = False,
+    ) -> Iterator[None]:
+        del num_nodes  # unused
+        regions = cls.regions_with_offering(instance_type,
+                                            accelerators,
+                                            use_spot,
+                                            region=region,
+                                            zone=None)
+        for r in regions:
+            assert r.zones is None, r
+            yield r.zones
+
+    def instance_type_to_hourly_cost(self,
+                                     instance_type: str,
+                                     use_spot: bool,
+                                     region: Optional[str] = None,
+                                     zone: Optional[str] = None) -> float:
+        return service_catalog.get_hourly_cost(instance_type,
+                                               use_spot=use_spot,
+                                               region=region,
+                                               zone=zone,
+                                               clouds='cudo')
+
+    def accelerators_to_hourly_cost(self,
+                                    accelerators: Dict[str, int],
+                                    use_spot: bool,
+                                    region: Optional[str] = None,
+                                    zone: Optional[str] = None) -> float:
+        del accelerators, use_spot, region, zone  # unused
+        return 0.0
+
+    def get_egress_cost(self, num_gigabytes: float) -> float:
+        # Change if your cloud has egress cost. (You can do this later;
+        # `return 0.0` is a good placeholder.)
+        return 0.0
+
+    def __repr__(self):
+        return 'Cudo'
```

Comment:
nit: this should be fine to remove, considering we have the `self._REPR` set already, and the `Cloud.__repr__` should be able to handle this.